internal/deregistration/deregistration.go

```diff
@@ -0,0 +1,50 @@
+package deregistration
+
+import (
+	"git.sr.ht/~spc/go-log"
+	"github.com/jakub-dzon/k4e-device-worker/internal/configuration"
+	"github.com/jakub-dzon/k4e-operator/models"
+	"github.com/jakub-dzon/k4e-device-worker/internal/workload"
+)
+
+type Deregistration struct {
```

Comment:
IMO that should be part of `registration.Registration` struct - the methods should belong to it.